include/inexor/vulkan-renderer/wrapper/mesh_buffer.hpp

```diff
@@ -26,60 +28,95 @@ namespace inexor::vulkan_renderer::wrapper {
 class MeshBuffer {
 
 private:
+    
     const wrapper::Device &m_device;
-    const std::string m_name;
+    std::string m_name;
+
     GPUMemoryBuffer m_vertex_buffer;
+
     std::optional<GPUMemoryBuffer> m_index_buffer;
+
     std::uint32_t m_number_of_vertices{0};
     std::uint32_t m_number_of_indices{0};
 
     // Don't forget that index buffers are optional!
-    bool m_index_buffer_available = false;
+    bool index_buffer_available = false;
 
 public:
-    /// @brief Creates a new vertex buffer and an associated index buffer.
-    MeshBuffer(const wrapper::Device &device, VkQueue data_transfer_queue,
+    // Delete the copy constructor so mesh buffers are move-only objects.
+    MeshBuffer(const MeshBuffer &) = delete;
+    MeshBuffer(MeshBuffer &&buffer) noexcept;
+
+    // Delete the copy assignment operator so uniform buffers are move-only objects.
+    MeshBuffer &operator=(const MeshBuffer &) = delete;
+    MeshBuffer &operator=(MeshBuffer &&) noexcept = default;
+
+    /// @brief Creates a new vertex buffer with an associated index buffer and copies memory into it.
+    MeshBuffer(const wrapper::Device &device, const VkQueue data_transfer_queue,
                const std::uint32_t data_transfer_queue_family_index, const VmaAllocator vma_allocator,
                const std::string &name, const VkDeviceSize size_of_vertex_structure,
                const std::size_t number_of_vertices, void *vertices, const VkDeviceSize size_of_index_structure,
                const std::size_t number_of_indices, void *indices);
 
-    /// @brief Creates a vertex buffer without index buffer.
-    MeshBuffer(const wrapper::Device &device, VkQueue data_transfer_queue,
+    /// @brief Creates a new vertex buffer with an associated index buffer but does not copy memory into it.
+    /// This is useful when you know the size of the buffer but you don't know it's data values yet.
+    MeshBuffer(const wrapper::Device &device, const VkQueue data_transfer_queue,
```

Comment:
We do not use `const` for parameters in header. (for values ofc)

Reply:
I will deal with this in another PR: https://github.com/inexorgame/vulkan-renderer/issues/239.